src/declarativedbusinterface.cpp

```diff
@@ -415,7 +416,8 @@ void DeclarativeDBusInterface::componentComplete()
 
 void DeclarativeDBusInterface::pendingCallFinished(QDBusPendingCallWatcher *watcher)
 {
-    QJSValue callback = m_pendingCalls.take(watcher);
+    QPair<QJSValue, QJSValue> callbacks = m_pendingCalls.take(watcher);
+    QJSValue callback = callbacks.first;
```

Comment:
Move declaration of callback & validity check to after error handling, line 437 or so.

Now errors can in theory be missed with a bad callback, which should not happen as the callability of the callback is checked when the D-Bus call is initiated.